File: Practice/src/exercise/OurUniquePassword.js
function solution(s, skip, index) {
    const c = [];
    let temp = 0;
    for (let i = 0; i < 26; i++) {
        for (let count = 1, minus = 0; count - minus <= index; count++) {
            temp = String.fromCharCode(((i + count) % 26) + 97);
            if (skip.includes(temp)) minus++;
        }
        c.push(temp);
    }
    return [...s].map((v) => c[v.charCodeAt(0) - 97]).join('');
    // const alphabet = [...'abcdefghijklmnopqrstuvwxyz'].filter(
    //     (c) => !skip.includes(c)
    // );
    // return [...s]
    //     .map((c) => alphabet[(alphabet.indexOf(c) + index) % alphabet.length])
    //     .join('');
}

File: Practice/src/exercise/OurUniquePassword.java
package exercise;

public class OurUniquePassword {
//	둘만의 암호
	public String solution(String s, String skip, int index) {
		StringBuilder answer = new StringBuilder();
		char[] c = new char[26];
		int temp = 0;
		for (int i = 0; i < 26; i++) {
			for (int count = 1, minus = 0; count - minus <= index; count++) {
				temp = (i + count) % 26 + 97;
				if (skip.indexOf(temp) > -1) {
					minus++;
				}
			}
			c[i] = (char) temp;
		}
		for (int i = 0; i < s.length(); i++) {
			answer.append(c[s.charAt(i) - 97]);
		}
		return answer.toString();
	}
}